### PR TITLE
feat: add waitForXPath to ElementHandle

### PR DESCRIPTION
**Pull request by OrKoN**
This PR adds waitForXPath to ElementHandle similar to waitForSelector.

### CODE REVIEW

File: src/common/JSHandle.ts [hunk 1]
```diff
@@ -416,6 +416,84 @@ export class ElementHandle<
     return result;
   }
 
+  /**
+   * Wait for the `xpath` within the element. If at the moment of calling the
+   * method the `xpath` already exists, the method will return immediately. If
+   * the `xpath` doesn't appear after the `timeout` milliseconds of waiting, the
+   * function will throw.
+   *
+   * If `xpath` starts with `//` instead of `.//`, the dot will be appended automatically.
```
Comment (jrandolf) [inline src/common/JSHandle.ts:425]: Is this comment necessary? I think it will just confuse people as almost everyone will use `//` for the start. We could also rephrase this to state `//` and `.//` are supported, without providing implementation details.

**Review by jrandolf-google — APPROVED** [review]

File: src/common/JSHandle.ts (hunk 1 above)
Comment (OrKoN) [reply src/common/JSHandle.ts:425]: `//` means querying from the root node so we cannot allow `//` as it might return elements outside of the element's subtree. So I think this comment is useful to indicate that we rewrite `//` as `.//`.

Comment (OrKoN) [reply src/common/JSHandle.ts:425]: Maybe I could phrase it better 🤔

Comment (jrandolf) [reply src/common/JSHandle.ts:425]: It's just a nit. Feel free to move forward with this.
